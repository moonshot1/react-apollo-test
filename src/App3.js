import React, { Component } from 'react'
import PropTypes from 'prop-types';

import { Query } from 'react-apollo';
import gql from 'graphql-tag';

export default class App3 extends Component {

  constructor(props){
    super(props);
    this.testMethod = this.testMethod.bind();
  }

  testMethod() {
    console.log("it runs");
  }

  render() {
    

    const QUERY = gql`
      query HeroFriends($episode: Episode!) {
        hero(episode: $episode) {
          name
        }
      }
    `;

    const HeroAndFriends = ({ episode }) => (
      <Query query={QUERY} variables={{ episode }}>
        {({ data, error, loading }) => {
          if (error) return 'Oops!';
          if (loading) return 'Patience young grasshopper...';
          // console.log("data obj", data);
          return (
            <div>
              <h1>{data.hero.name}</h1>

            </div>
          );
        }}
      </Query>
    )
    this.testMethod();
    return (
      <div>
        {HeroAndFriends({ episode: 'EMPIRE' })}
      </div>
    )
  }
}
