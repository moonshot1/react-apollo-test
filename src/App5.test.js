import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import App3 from './App3';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('App4', () => {
  it('should render correctly', () => {
    const output = shallow(
      <App3 />
    );

    console.log(output.instance());
    //console.log(output.instance().dummyMethod);

    expect(shallowToJson(output)).toMatchSnapshot();
  });
});