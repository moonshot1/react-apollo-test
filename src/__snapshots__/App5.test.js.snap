// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App4 should render correctly 1`] = `
<div>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "HeroFriends",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "episode",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "episode",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "hero",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Episode",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "episode",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 118,
          "start": 0,
        },
      }
    }
    variables={
      Object {
        "episode": "EMPIRE",
      }
    }
  />
</div>
`;
